sklearn/preprocessing/tests/test_function_transformer.py

```diff
@@ -156,15 +158,92 @@ def test_check_inverse():
         trans = FunctionTransformer(func=np.expm1,
                                     inverse_func=np.log1p,
                                     accept_sparse=accept_sparse,
-                                    check_inverse=True)
+                                    check_inverse=True,
+                                    validate=True)
         Xt = assert_no_warnings(trans.fit_transform, X)
         assert_allclose_dense_sparse(X, trans.inverse_transform(Xt))
 
     # check that we don't check inverse when one of the func or inverse is not
     # provided.
     trans = FunctionTransformer(func=np.expm1, inverse_func=None,
-                                check_inverse=True)
+                                check_inverse=True, validate=True)
     assert_no_warnings(trans.fit, X_dense)
     trans = FunctionTransformer(func=None, inverse_func=np.expm1,
-                                check_inverse=True)
+                                check_inverse=True, validate=True)
     assert_no_warnings(trans.fit, X_dense)
+
+
+@pytest.mark.parametrize(
+    "X",
+    [np.array([[0, 1], [2, 3]]),
+     np.array([[0, 1], [2, np.nan]]),
+     np.array([[0, 1], [2, np.inf]]),
+     np.array([[0, 1], [np.inf, np.nan]])]
+)
+@pytest.mark.parametrize(
+    "force_all_finite",
+    [True, False, 'allow-nan']
+)
+def test_function_transformer_finiteness_pandas(X, force_all_finite):
+    pd = pytest.importorskip('pandas')
+    X_df = pd.DataFrame(X)
+
+    def func(X):
+        return X.columns
+
+    transformer = FunctionTransformer(force_all_finite=force_all_finite,
+                                      validate=True)
+
+    should_fail = False
+    if force_all_finite is True:
+        if not np.isfinite(X).all():
+            should_fail = True
+    elif force_all_finite == 'allow-nan':
+        if np.isinf(X).any():
+            should_fail = True
+
+    if should_fail:
+        with pytest.raises(ValueError, match="Input contains"):
+            transformer.fit_transform(X_df)
+    else:
+        transformer.fit_transform(X_df)
+
+
+@pytest.mark.parametrize(
+    "is_dataframe",
+    [True, False]
```

Comment:
nitpick: I know that's the code style in pytest examples, but do you think it's really worth spreading something that can be written in 1 line over 4 lines? I'm not convinced it helps readability..